CONTRIBUTING.rst

```diff
@@ -328,6 +334,45 @@ detailing the test coverage::
   networkx/algorithms/approximation/clique.py         42      1     18      1    97%
   ...
 
+Adding tests
+------------
+
+If you're **new to testing**, see existing test files for examples of things to do.
+**Don't let the tests keep you from submitting your contribution!**
+If you're not sure how to do this or are having trouble, submit your pull request
+anyway.
+We will help you create the tests and sort out any kind of problem during code review.
+
+General guidelines for writing good tests:
+
+- doctests always assume ``import networkx as nx`` so don't add that
+- avoid test classes when possible
+- use the following decorators (``@pytest.mark.parametrize``, ``@pytest.mark.skipif``, ``@pytest.mark.xfail``, ``@pytest.mark.slow``)
+- use ``pytest.importskip`` for numpy, scipy, pandas, and matplotlib b/c of PyPy.
+- something about ``networkx/conftest.py``
```

Comment:
I think most users don't have to interact with/modify conftest.py when adding new tests, so I'm not sure it needs to be explicitly mentioned here.

Reply:
What about if the file uses numpy? Or if we deprecate a function?  I'm fine if we just decide to do that for people. But where is the borderline? :}

At some point the contributing document will likely be too much for first timers. But it seems like too much is maybe better than too little.  Maybe we can split off a section for first timers and then have all the details somewhere else. :)

Reply:
I moved this to the reference guide and I also edited based on your comments.